content/en/blog/_posts/2020-03-30-topology-manager-beta.md

```diff
@@ -0,0 +1,517 @@
+1. A brief introduction to NUMA and why it is important
+2. The policies available to end-users to ensure NUMA alignment of CPUs and devices
+3. The internal details of how the **<code>TopologyManager</code>** works
+4. Current limitations of the <strong><code>TopologyManager</code></strong>
+5. Future directions of the <strong><code>TopologyManager</code></strong>
+
+## So, what is NUMA and why do I care?
+
+The term NUMA stands for Non-Uniform Memory Access. It is a technology available on multi-cpu systems that allows different CPUs to access different parts of memory at different speeds. Any memory directly connected to a CPU is considered “local” to that CPU and can be accessed very fast. Any memory not directly connected to a CPU is considered “non-local” and will have variable access times depending on how many interconnects must be passed through in order to reach it. On modern systems, the idea of having “local” vs. “non-local” memory can also be extended to peripheral devices such as NICs or GPUs. For high performance, CPUs and devices should be allocated such that they have access to the same local memory.
+
+All memory on a NUMA system is divided into a set of “NUMA nodes”, with each node representing the local memory for a set of CPUs or devices. We talk about an individual CPU as being part of a NUMA node if its local memory is associated with that NUMA node. 
+
+We talk about a peripheral device as being part of a NUMA node based on the shortest number of interconnects that must be passed through in order to reach it.
+
+For example, in Figure 1, CPUs 0-3 are said to be part of NUMA node 0, whereas CPUs 4-7 are part of NUMA node 1. Likewise GPU 0 and NIC 0 are said to be part of NUMA node 0 because they are attached to Socket 0, whose CPUs are all part of NUMA node 0. The same is true for GPU 1 and NIC 1 on NUMA node 1.
+
+
+![drawing](/images/blog/2020-03-25-kubernetes-1.18-release-announcement/example-numa-system.png)
+
+        Figure 1: An example system with 2 NUMA nodes, 2 Sockets with 4 CPUs each, 2 GPUs, and 2 NICs. CPUs on Socket 0, GPU 0, and NIC 0 are all part of NUMA node 0. CPUs on Socket 1, GPU 1, and NIC 1 are all part of NUMA node 1. 
+
+Although the example above shows a 1-1 mapping of NUMA Node to Socket, this is not necessarily true in the general case. There may be multiple sockets on a single NUMA node, or individual CPUs of a single socket may be connected to different NUMA nodes. Moreover, emerging technologies such as Sub-NUMA Clustering ([available on recent intel CPUs](https://software.intel.com/en-us/articles/intel-xeon-processor-scalable-family-technical-overview)) allow single CPUs to be associated with multiple NUMA nodes so long as their memory access times to both nodes are the same (or have a negligible difference).
+
+The **<code>TopologyManager</code>** has been built to handle all of these scenarios.
+
+
+![drawing](/images/blog/2020-03-25-kubernetes-1.18-release-announcement/align-up.png)
```

Comment:
Let's go ahead and remove this image, and just replace it with the heading:
```
## Align Up! It's a TeaM Effort! ##
```
It looks more awkward here than it did in the google doc since it doesn't match the formatting.

Reply:
It looks like this has been fixed, looks ok to me.